s/core_test.go

```diff
@@ -0,0 +1,193 @@
+package s_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/capatazlib/go-capataz/c"
+	"github.com/capatazlib/go-capataz/s"
+)
+
+func ExampleNew() {
+	// Build a supervision tree with two branches
+	rootSpec := s.New(
+		"root",
+		// first sub-tree
+		s.WithSubtree(
+			s.New("sub-tree-1",
+				s.WithChildren(
+					c.New("child-1", func(ctx context.Context) error {
+						<-ctx.Done()
+						return nil
+					}),
+					c.New("child-2", func(ctx context.Context) error {
+						<-ctx.Done()
+						return nil
+					}),
+				),
+			),
+		),
+		// second sub-tree
+		s.WithSubtree(
+			s.New("sub-tree-2",
+				s.WithChildren(
+					c.New("child-3", func(ctx context.Context) error {
+						<-ctx.Done()
+						return nil
+					}),
+					c.New("child-4", func(ctx context.Context) error {
+						<-ctx.Done()
+						return nil
+					}),
+				),
+			),
+		),
+	)
+
+	// Spawn goroutines of supervision tree
+	sup, err := rootSpec.Start(context.Background())
+	if err != nil {
+		fmt.Printf("Error starting system: %v\n", err)
+	}
+	// Wait for supervision tree to exit, this happens only if the errors are not
+	// recoverable
+	err = sup.Wait()
+	if err != nil {
+		fmt.Printf("Supervisor failed with error: %v\n", err)
+	}
+}
+
+// Test a supervision tree with a single child starts and stops
+func TestStartSingleChild(t *testing.T) {
+	cs := waitDoneChild("one")
+
+	events, err := observeSupervisor(
+		"root",
+		[]s.Opt{
+			s.WithChildren(cs),
+		},
+		noWait,
+	)
+	assert.Nil(t, err)
+
+	assertExactMatch(t, events,
```

Comment:
This point needs clarification, the supervisor root is responsible of all it's associated children. In this context the supervisor has a fixed set of children (static tree via the SupervisorSpec), it's a Supervisor's job to ensure all it's children are spawned (in the right order) before it can do any restart operations. I'll add some comments in the code that clarify the roles of the Supervisor so that it becomes clearer why supervisor root is at the end of the start events group

Reply:
I added [some new documentation](https://github.com/capatazlib/go-Capataz/blob/807c5b6e4cb5811d104c58f447d5e82bee2c81f5/s/core.go#L252) that explains this in detail